academy/models/teachers.py

```diff
@@ -0,0 +1,8 @@
+from odoo import models, fields
+
+
+class Teacher(models.Model):
+    _name = 'academy.teachers'
```

Comment:
this is still not changed, it has to be teacher not teachers

```suggestion
    _name = 'academy.teacher'
```